main.cpp

```diff
@@ -53,8 +55,9 @@ struct U
     {   
         if(updatedValue != nullptr)
         {
+            float &newValue = *updatedValue;
```

Comment:
You can convert the function arguments to references.  You can't convert the return type.
- [ ] revise